### PR TITLE
Update to improved DumpResponse

### PR DESCRIPTION
**Pull request by nbramblett**
Updates our usage of DumpResponse away from the httputil implementation
(which is not recommended for live servers) and onto our own copy, which
has better support for sizing byte buffers.

### CODE REVIEW

File: httpcache.go [hunk 1]
```diff
@@ -586,3 +585,82 @@ func NewMemoryCacheTransport() *Transport {
...
+// It returns an error if the initial slurp of all bytes fails. It does not attempt
+// to make the returned ReadClosers have identical error-matching behavior.
+func drainBody(b io.ReadCloser, expectedLength int64) (r1, r2 io.ReadCloser, err error) {
+	if b == nil || b == http.NoBody {
+		// No copying needed. Preserve the magic sentinel meaning of NoBody.
+		return http.NoBody, http.NoBody, nil
+	}
+	var buf = &bytes.Buffer{}
+	if expectedLength > 0 {
+		if _, err = io.CopyN(buf, b, expectedLength); err != nil {
```
Comment (robfig) [inline httpcache.go]: I don't think this will do it -- I'd suggest something like
```
if expectedLength > 0 {
  buf.Grow(expectedLength)
}

... 
```

Comment (robfig) [reply httpcache.go]: (CopyN just limits the amount copied. It operates on generic io.Writer so doesn't do anything special for bytes.Buffer)

Comment (nbramblett) [reply httpcache.go]: Fixed

**Review by robfig — APPROVED** [review]
Looks good! Should be able to easily see impact in the telemetry on # allocs

**Pull request merged**